inc/WpdbStorage/DataSourceCrud.php

```diff
@@ -36,8 +36,20 @@ public static function get_config_by_uuid( string $uuid ): array|WP_Error {
 		return new WP_Error( 'data_source_not_found', __( 'Data source not found', 'remote-data-blocks' ), [ 'status' => 404 ] );
 	}
 
-	public static function get_configs(): array {
-		return self::get_all_configs();
+	public static function get_configs(): array|WP_Error {
+		$configs = self::get_all_configs();
+		$valid_configs = [];
+
+		foreach ( $configs as $config ) {
+			$instance = self::inflate_config( $config );
+
+			// If the data source is valid, set a transient field called enabled to true or false otherwise.
+			$config['enabled'] = ! is_wp_error( $instance );
```

Comment:
I've added this new transient field whose sole purpose is to reflect if that data source should be enabled or not. Given that this method is called anytime a get is requested, regardless of it being a get all or get one (we should look to optimize this using our learnings from the workflow plugin), this will always be set.

I didn't add this in the `migrate` method as the output of the inflated config isn't returned and that method wouldn't be called for configs that aren't tied to a supported data source.